client/lib/abtest/index.js

```diff
@@ -133,28 +154,29 @@ ABTest.prototype.isEligibleForAbTest = function() {
 	const clientLanguage = client.language || client.userLanguage || 'en';
 	const clientLanguagesPrimary = ( client.languages && client.languages.length ) ? client.languages[ 0 ] : 'en';
 	const localeFromSession = i18n.getLocaleSlug() || 'en';
-	const englishMatcher = /^en-?/i;
 
 	if ( ! store.enabled ) {
 		debug( '%s: Local storage is not enabled', this.experimentId );
 		return false;
 	}
 
-	if ( ! this.allowAnyLocale ) {
-		if ( isUserSignedIn() && user.get().localeSlug !== 'en' ) {
-			debug( '%s: User has a non-English locale', this.experimentId );
+	if ( this.localeTargets ) {
```

Comment:
If you do switch to `any` you might have to instead `isArray` here or test explicitly `!== 'any'` or something.